stdlib/int16.gr

```diff
@@ -0,0 +1,445 @@
+/**
+ * Utilities for working with the Int16 type.
+ * @example include "int16"
+ *
+ * @since v0.6.0
+ */
+
+module Int16
+
+include "runtime/unsafe/wasmi32"
+include "runtime/exception"
+include "runtime/numbers"
+from Numbers use { tagSimple, coerceNumberToShortInt }
+include "runtime/unsafe/constants"
+from Constants use { _SMIN16_I32, _SMAX16_I32, _SMIN16_I64, _SMAX16_I64 }
+include "runtime/dataStructures"
+from DataStructures use { tagInt16, untagInt16 }
+
+@unsafe
+let _DATA_OFFSET = 8n
+@unsafe
+let _TAG_BYTE = 0b10010n
+@unsafe
+let _DATA_MASK = 0xffffff00n
+
+/**
+ * Converts a Number to an Int16.
+ *
+ * @param number: The value to convert
+ * @returns The Number represented as an Int16
+ *
+ * @since v0.6.0
+ */
+@unsafe
+provide let fromNumber = (number: Number) => {
+  let val = coerceNumberToShortInt(
+    number,
+    _SMIN16_I32,
+    _SMAX16_I32,
+    _SMIN16_I64,
+    _SMAX16_I64,
+    false
+  )
+  tagInt16(val)
```

Comment:
I'm wondering if we should expose this (and co) from the runtime/number file as `coerceNumberToInt16`, etc. Mostly just wondering because it'd keep things more consistent and we're going to eventually want all `@unsafe` stuff in the runtime.

Reply:
Sure I can do this; pretty much every function in the short int standard libraries are `@unsafe` though so that'll be a lot of code moved :laughing: